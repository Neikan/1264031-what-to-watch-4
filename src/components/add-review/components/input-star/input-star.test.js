// Импорт библиотек
import React from "react";
import renderer from "react-test-renderer";

// Импорт компонентов
import InputStar from "./input-star.jsx";


describe(`Test InputStar component`, () => {
  test(`InputStar component is created and rendered correctly when checked`, () => {
    const tree = renderer.create(
        <InputStar
          checked={false}
          countStar={1}
          onChange={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


  test(`InputStar component is created and rendered correctly when non-cheked`, () => {
    const tree = renderer.create(
        <InputStar
          checked={true}
          countStar={1}
          onChange={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
