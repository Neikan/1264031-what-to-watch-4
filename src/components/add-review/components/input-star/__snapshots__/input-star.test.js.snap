// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputStar component InputStar component is created and rendered correctly when checked 1`] = `
Array [
  <input
    checked={false}
    className="rating__input"
    id="star-1"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={1}
  />,
  <label
    className="rating__label"
    htmlFor="star-1"
  >
    Rating 1
  </label>,
]
`;

exports[`Test InputStar component InputStar component is created and rendered correctly when non-cheked 1`] = `
Array [
  <input
    checked={true}
    className="rating__input"
    id="star-1"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={1}
  />,
  <label
    className="rating__label"
    htmlFor="star-1"
  >
    Rating 1
  </label>,
]
`;
